src/Features/Core/Portable/PopulateSwitch/AbstractPopulateSwitchDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,121 @@
+﻿using Microsoft.CodeAnalysis.Shared.Extensions;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Threading;
+
+namespace Microsoft.CodeAnalysis.Diagnostics.PopulateSwitch
+{
+    internal abstract class AbstractPopulateSwitchDiagnosticAnalyzerBase<TLanguageKindEnum> : DiagnosticAnalyzer, IBuiltInAnalyzer where TLanguageKindEnum : struct
+    {
+        private static readonly LocalizableString s_localizableTitle = new LocalizableResourceString(nameof(FeaturesResources.AddMissingStatements), FeaturesResources.ResourceManager, typeof(FeaturesResources));
+        private static readonly LocalizableString s_localizableMessage = new LocalizableResourceString(nameof(WorkspacesResources.PopulateSwitch), WorkspacesResources.ResourceManager, typeof(WorkspacesResources));
+
+        private static readonly DiagnosticDescriptor s_descriptor = new DiagnosticDescriptor(IDEDiagnosticIds.PopulateSwitchDiagnosticId,
+                                                                    s_localizableTitle,
+                                                                    s_localizableMessage,
+                                                                    DiagnosticCategory.EditAndContinue,
+                                                                    DiagnosticSeverity.Warning,
```

Comment:
This is an analyzer, so I believe it can be turned off.  Additionally, Dustin Campbell told me he does not like hidden diagnostics, so I changed it.